pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -595,9 +595,85 @@ func getSubnetIDForLB(compute *gophercloud.ServiceClient, node v1.Node) (string,
 	return "", ErrNotFound
 }
 
+// applyNodeSecurityGroupIDForLB associates the security group with all the ports on the nodes.
+func applyNodeSecurityGroupIDForLB(compute *gophercloud.ServiceClient, network *gophercloud.ServiceClient, nodes []*v1.Node, sg string) error {
+	for _, node := range nodes {
+		nodeName := types.NodeName(node.Name)
+		srv, err := getServerByName(compute, nodeName)
+		if err != nil {
+			return err
+		}
+
+		listOpts := neutronports.ListOpts{
```

Comment:
I am thinking, should we make getPorts() function which takes neutronports.listopts as parameter and returns port object. We could use that in another function as well in future if needed

so rows 607-617 could be removed and move to own function